src/DeadRegions/Program.cs

```diff
@@ -0,0 +1,290 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.DeadRegionAnalysis;
+
+namespace DeadRegions
+{
+    internal class DeadRegions
+    {
+        private static readonly char[] s_symbolSeparatorChars = new[] { ';', ',' };
+
+        private static OptionParser s_options = new OptionParser();
+        private static AnalysisEngine s_engine;
+
+        private static IList<string> s_filePaths;
+        private static List<string> s_ignoredSymbols = new List<string>();
+        private static List<string> s_definedSymbols = new List<string>();
+        private static List<string> s_disabledSymbols = new List<string>();
+        private static List<IEnumerable<string>> s_symbolConfigurations = new List<IEnumerable<string>>();
+        private static Tristate s_undefinedSymbolValue = Tristate.Varying;
+
+        private static bool s_printDisabled;
```

Comment:
Mutable statics ...  why not put all of this global state into a Options / State type and pass it to the appropriate code paths?